docs/rfc/rfc-005-accounts-auth.md

```diff
@@ -0,0 +1,299 @@
+1. Recoverable and, in general, more intelligent accounts
+2. Streamlined extensions of authentication mechanisms (custom cryptographic pairings, but not only)
+3. Vesting accounts
+
+The proposal's full potential is realized when integrated with modules like `x/wasm` and `x/evm`. These modules facilitate 
+the deployment of accounts with custom and dynamic code, eliminating the need for app-level definition of account code. This enables the
+proposed account abstraction's use cases to evolve with the network itself.
+
+## Proposal
+
+
+### Account role
+
+The proposal begins by clarifying the role of an account within the Cosmos SDK's runtime. Like modules, accounts are
+actors that can send and receive messages. They modify each other's states through state transitions, which are represented
+as messages. The runtime does not differentiate between actors or apply any authorization logic; it simply ensures that
+sender actors cannot impersonate others.
+
+This leads to significant implications: Externally Owned Accounts (EOA) and ModuleAccounts (system-owned identities) can
+be treated as the same entity. Authorization, which determines which actor can execute a particular state transition
+towards another actor, must be managed by the recipient of the state transition. For example, the 'bank' module decides
+if the 'mint' module is permitted to inflate a coin's supply. This eliminates the need for permissions in the 'auth' 
+module as they currently exist.
+
+The key difference between EOA and ModuleAccount lies in the origin of state transitions on their behalf. In the case of
+ModuleAccount, a state transition cannot be initiated from a transaction (TX), as no valid credential can prove the sender
+is the ModuleAccount itself.
+
+We can distinguish between EOA and ModuleAccount by considering an account as a ModuleAccount (or SystemOwnedAccount)
+when it is not possible to initiate a state transition on its behalf from a TX. Given that this distinction occurs during
+the VerifySignature stage of the AnteHandler and that authorization is managed by the state transition recipient, the
+execution runtime or other system actors to manage differences between account types. Instead, they should be treated as
+business logic units capable of sending and receiving messages. This further proves that there's no need to distinguish
+`ModuleAccounts` from `EOA`.
+
+### Account Definition
+
+We define the new `Account` type, which is what an account needs to implement to be treated as such.
+An `Account` type is defined at APP level, so it cannot be dynamically loaded as the chain is running without upgrading the 
+node code, unless we use something like `x/wasm`. 
+
+```go
+type InternalAccount struct {
+	init    func(ctx *Context, msg proto.Message) (*InitResponse, error)
+	execute func(ctx *Context, msg proto.Message) (*ExecuteResponse, error)
+	query   func(ctx *Context, msg proto.Message) (proto.Message, error)
+	schema  func() *Schema
```

Comment:
Do you have a definition of the `Schema` object anywhere?

Will there need to be disambiguate the name from the [orm](https://github.com/cosmos/cosmos-sdk/blob/main/orm/model/ormtable/table.go#L147-L152) or [collections](https://github.com/cosmos/cosmos-sdk/blob/main/collections/schema.go#L128-L133) versions?

Should it be an interface instead?